src/state/user/reducer.ts

```diff
@@ -61,12 +67,50 @@ export interface UserState {
   showSurveyPopup: boolean | undefined
 
   showDonationLink: boolean
+
+  // mod, favourite tokens
+  favouriteTokens: {
+    [chainId: number]: {
+      [address: string]: SerializedToken
+    }
+  }
 }
 
 function pairKey(token0Address: string, token1Address: string) {
   return `${token0Address};${token1Address}`
 }
 
+function serializeToken(token: Token): SerializedToken {
```

Comment:
Repeated fn with the same name and similar implementation.

See src/state/user/hooks